### ISSUE TITLE
Change whitelist and blacklist terms

### ISSUE THREAD

**Issue by NicoleSchwartz** [optiopay/klar#171]
Hello!

Although currently common terms whitelist and blacklist, these terms are hurtful to some people. I'd like to propose a change to equally descriptive terms Allowlist and Denylist - see other projects changing here https://github.com/rails/rails/issues/33677

It could be done as a deprecation where for a (extended) period of of time both terms could be used (a mapping) until a specific version/date where only the new terms would work.

**Comment by blacklistisnotracist** [comment]
@NicoleSchwartz https://www.youtube.com/watch?v=GNXm7juuM-8

**Comment by NicoleSchwartz** [comment]
@blacklistisnotracist - A few months ago, an NCSC customer contacted me to ask if we would consider making a small but significant change to some of the wording we use on the NCSC website. When she asked the question, I immediately smacked myself in the head for not thinking of it a long time ago. And I was really glad to say: yes, we will make this change straight away, and I'm sorry you had to come and ask us to do it.

It's fairly common to say whitelisting and blacklisting to describe desirable and undesirable things in cyber security. For instance, when talking about which applications you will allow or deny on your corporate network; or deciding which bad passwords you want your users not to be able to use.

However, there's an issue with the terminology. It only makes sense if you equate white with 'good, permitted, safe' and black with 'bad, dangerous, forbidden'. There are some obvious problems with this. So in the name of helping to stamp out racism in cyber security, we will avoid this casually pejorative wording on our website in the future. No, it's not the biggest issue in the world - but to borrow a slogan from elsewhere: every little helps.

You may not see why this matters. If you're not adversely affected by racial stereotyping yourself, then please count yourself lucky. For some of your colleagues (and potential future colleagues), this really is a change worth making. From now on, the NCSC will use 'allow list' and 'deny list' in place of 'whitelist' and 'blacklist' on our website. Which, in fact, is clearer and less ambiguous. So as well as being more inclusive of all, this is a net benefit to our web content. We are editing our guidance across the website to update the terms, but if you do spot any in the meantime then please do contact us.

I hope that if you're seeking to make this, or similar changes in your own organisation, this blog post helps you to convince people around you that it's worth doing. And finally, a word from the NCSC's Technical Director Ian Levy (supported by the full NCSC Management Board): "If you’re thinking about getting in touch saying this is political correctness gone mad, don’t bother."



Emma W
Head of Advice and Guidance, NCSC
https://www.ncsc.gov.uk/blog-post/terminology-its-not-black-and-white

**Comment by blacklistisnotracist** [comment]
@NicoleSchwartz 

> no one is taking issue with the word black

Wrong. The only reason people want to ban the word blacklist is because blacklist has negative connotations and has "black", which is also used to refer to a group of people coincidentally, in it so it could be potentially offensive to black people.

Your premise is "blacklist could be potentially offensive for black people." But is this really true? Based on what? If your premise is false, you can draw whatever conclusion you want.

Also, a few number of white people said "blacklist carries racist connotations" and justify the prohibition of blacklist based on this false argument. Again, blacklist doesn't have any racist origins, and it is universally agreed that it has nothing to do with race or racism.

This is as ridiculous as calling yellow cards in soccer is a racist word and could potentially offense yellow people.

**Comment by Ro0t-OS** [comment]
@blacklistisnotracist I get what you're trying to say, I really do, but I don' t think you really get it. I, as a conscious black female red teamer, get really annoyed by the way some syntax in infosec has been 'developed'. We, as black ethical hackers historically didn't have a say and kind of still don't. What's being discussed here is specific to the terms blacklist and whitelist. Not yellow and every other color you're trying to bring in (that's being very 'all-lives/colors-matter-ish'). I'm not sure if you know this but the InfoSec/Developer field has been predominantly white males historically and still is today sadly. Being that a lot of tech terms were developed by white men with unconscious/conscious biases, I'm not surprised in the least that they didn't come up with SilverList or TurquoiseList. I understand not wanting to change something because of how its always been but that's simply not an excuse to not make our syntax systems better (as we do with our code/tools) and have some real conversations around it. It's really not that difficult. Based on the conversation above, it appears your very passive in your counter arguments. Also, not all words are offensive. You stated that people can get offended by any words. You should be more precise in saying that people get offended by flat out ignorant/stereotypical words that have been used historically to hurt a group of people. This really isn't that hard to grasp in my opinion/experience as a black woman. Lastly, white people aren't the only ones that claim this. Do you speak for the whole white race or something? If so, ya'll need to sit and have some unconscious bias training. I've definitely heard from other black people in the ethical hacking space about infosec's usage of some of the words. Not all. So in conclusion, critical thinking is not limited to what we've been taught or believe. It also applies to 'what could be'. Growth of the mind and growth of the systems/infosec/dev field is a choice. Those that fight against change and growth will merely be in conflict with themselves and continue thriving in their echo chambers.

**Comment by blacklistisnotracist** [comment]
@Ro0t-OS 

Well, any word can offend some people. This is true.
https://www.youtube.com/watch?v=gEOhG8I-Bu4

Do you associate your race with the color black? Black carries negative connotations anyway because of its association with absence of light. This had been so from the very beginning of our history, even before humans started labeling people with colors. So, in short, black color is not yours. You should be questioning your label instead of trying to get rid of all negative words that have black in it.

If you can't find it ridiculous to stop using red color for Native Americans, then you don't have the right to speak for blacklist. It's not about all-lives-matter shit against black-lives-matter. It's not one of those whataboutism instances. I'm just pointing out how ridiculous it looks.

If we allow this kind of removal of any negative black* words even though they have nothing to do with race, our culture will be dead. Our language will be dead. What do you want to achieve here? Do you want to watch a horror movie without any dark scene?

To be honest, if you ever find any racial connotations in white/black color, you're a racist. Nobody really think of skin color when they see a compound word that has a color in it.

**Comment by Ro0t-OS** [comment]
@blacklistisnotracist Yikes.....you legit don't get it and aren't aware that you don't get it. This is where I stop. You can't speak with a closed mind or wall. Best of luck with your continued success (?) in showing you don't get it.

**Comment by blacklistisnotracist** [comment]
So your premise is blacklist is a racist word, right? Then, I would gently disagree with that. It doesn't have racist connotations, it never has been used as discriminative word against black people.

Would you mind explaining what I didn't get more in details?

**Comment by blacklistisnotracist** [comment]
Anyway, I'm not opposing to the change to allowlist/denylist. I just wanted to point out that blacklist doesn't have racist origins and has never been an abusive, discriminative word against black people. If you think black color=bad is wrong, then, that's where I can't stand. Black color has nothing to do with black people.
